File: generated-code/ruby/codecraft/FileReadWrite/model/move_action.rb
require './vec2_int'

module Model

# Move action
class MoveAction
    # Target position
    attr_accessor :target
    # Whether to try find closest position, if path to target is not found
    attr_accessor :find_closest_position
    # Whether to destroy other entities on the way
    attr_accessor :break_through

    def initialize(target, find_closest_position, break_through)
        @target = target
        @find_closest_position = find_closest_position
        @break_through = break_through
    end

    # Read MoveAction from input stream
    def self.read_from(stream)
        target = Vec2Int.read_from(stream)
        find_closest_position = stream.read_bool()
        break_through = stream.read_bool()
        MoveAction.new(target, find_closest_position, break_through)
    end

    # Write MoveAction to output stream
    def write_to(stream)
        @target.write_to(stream)
        stream.write_bool(@find_closest_position)
        stream.write_bool(@break_through)
    end

    def to_s
        string_result = "MoveAction { "
        string_result += "target: "
        string_result += @target.to_s
        string_result += ", "
        string_result += "find_closest_position: "
        string_result += @find_closest_position.to_s
        string_result += ", "
        string_result += "break_through: "
        string_result += @break_through.to_s
        string_result += " }"
        string_result
    end

    def to_str
        to_s
    end
end

end

File: generated-code/php/codecraft/FileReadWrite/Model/MoveAction.php
<?php

namespace Model {
    require_once 'Stream.php';
    require_once 'Vec2Int.php';

    /**
     * Move action
     */
    class MoveAction
    {
        /**
         * Target position
         */
        public \Vec2Int $target;
        /**
         * Whether to try find closest position, if path to target is not found
         */
        public bool $findClosestPosition;
        /**
         * Whether to destroy other entities on the way
         */
        public bool $breakThrough;
    
        function __construct(\Vec2Int $target, bool $findClosestPosition, bool $breakThrough)
        {
            $this->target = $target;
            $this->findClosestPosition = $findClosestPosition;
            $this->breakThrough = $breakThrough;
        }
    
        /**
         * Read MoveAction from input stream
         */
        public static function readFrom(\InputStream $stream): MoveAction
        {
            $target = \Vec2Int::readFrom($stream);
            $findClosestPosition = $stream->readBool();
            $breakThrough = $stream->readBool();
            return new MoveAction($target, $findClosestPosition, $breakThrough);
        }
        
        /**
         * Write MoveAction to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $this->target->writeTo($stream);
            $stream->writeBool($this->findClosestPosition);
            $stream->writeBool($this->breakThrough);
        }
    }
}